src/tools/ceph_dedup_tool.cc

```diff
@@ -1028,6 +1281,9 @@ int main(int argc, const char **argv)
 	     op_name == "chunk-repair" ||
 	     op_name == "dump-chunk-refs") {
     return chunk_scrub_common(opts, args);
+  } else if (op_name == "chunk-dedup" ||
+	     op_name == "object-dedup") {
```

Comment:
I don't think I understand what these two operations are supposed to do.  Where would a user find that information?

Reply:
The reason why I added two operations here is to make dedup easy to use. To use dedup currently. there are two cases using rados API as belows:
1. set_chunk:
 a. read chunk, b. create a chunk object on the lower pool, c. generate fingerprint to be used as OID, d. call set_chunk
2. tier_flush:
 a. make the target object as a manifest object using set_chunk, b. call tier_flush, c. call tier_evict.

These two cases probably are complicated a bit for users to use deduplication. So, a helper function would be helpful.
We can add the usage of deduplication to document to let users know.

Users executes "ceph-dedup-tool estimate" to find out how many duplicate chunks the object has, then call either chunk-dedup or object-dedup if needed.

Am I missing something?

Reply:
Right, I'm saying that users need some kind of explanation for what these do.

Reply:
At the least, add a short comment at the top of the handler for each of those commands explaining what they do.